packages/repository/test/unit/repositories/constraint-utils.unit.ts

```diff
@@ -0,0 +1,85 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {FilterBuilder, Filter, Where, WhereBuilder} from '../../../src/query';
+import {
+  constrainFilter,
+  constrainWhere,
+} from '../../../src/repositories/constraint-utils';
+
+describe('constraint utility functions', () => {
+  let inputFilter: Filter = {};
+  let inputWhere: Where = {};
+
+  before(() => {
+    inputFilter = givenAFilter();
```

Comment:
I'd appreciate a comment here to let other people know what `inputFilter` and `inputWhere` looks like as objects

Reply:
The purpose of a test-data builder like `givenAFilter` is to create any valid filter, this generic value should be used only in places where the test does not really care about what filter is used.

For tests that where the actual filter is important (e.g. because assertions make assumptions about inputs), it's important to capture the important input data directly in the test. See also my earlier comment https://github.com/strongloop/loopback-next/pull/1383#discussion_r192046338 and the example in https://github.com/strongloop/loopback-next/pull/1383#discussion_r192048728 showing how to write a test that's using these test-data builders and yet makes it clear which part of the input filter is important and where the values used in assertions are coming from.

A slightly improved example:

```ts
const input = givenAFilterWithWhere({x: 'x'});
const constraint = givenAWhereConstraint({id: '10'});
const result = constrainFilter(input, constraint);
expect(result).to.containEql({
  where: {x: 'x', id: '10'}
});
```

My point is that if you feel we need a comment to explain what `inputFilter` and `inputWhere` looks like to make it easier to understand what the tests are doing, then we should improve the tests instead.